stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -115,7 +122,7 @@ internal class PaymentSheetViewModel internal constructor(
 
     fun fetchIsGooglePayReady() {
         if (isGooglePayReady.value == null) {
-            if (args.isGooglePayEnabled) {
+            if (args.isGooglePayEnabled && args.clientSecret is PaymentIntentClientSecret) {
```

Comment:
args.isGooglePayEnabled should not return true unless the clientSecret is PaymentIntentClientSecret